core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -253,12 +254,34 @@ public static void purgeTable(Catalog catalog, TableIdentifier ident) {
     }
   }
 
-  public static LoadTableResponse loadTable(Catalog catalog, TableIdentifier ident) {
+  private static TableMetadata getTableMetaData(
+      BaseTable baseTable, RESTSessionCatalog.SnapshotMode snapshotMode) {
+    TableMetadata tableMetadata = baseTable.operations().current();
+
+    switch (snapshotMode) {
+      case ALL:
+        return tableMetadata;
+      case REFS:
+        {
+          TableMetadata refsMetaData = TableMetadata.buildFrom(tableMetadata).build();
+          Set<Long> referencedSnapshotIds =
+              refsMetaData.refs().values().stream()
+                  .map(SnapshotRef::snapshotId)
+                  .collect(Collectors.toSet());
+          refsMetaData.removeSnapshotsIf(s -> !referencedSnapshotIds.contains(s.snapshotId()));
+          return refsMetaData;
+        }
+    }
+    return tableMetadata;
+  }
+
+  public static LoadTableResponse loadTable(
```

Comment:
I don't think we can just change a public method as that will break API compatibility: https://github.com/apache/iceberg/blob/master/CONTRIBUTING.md#semantic-versioning

@sandflee what's the expectation on your end for this PR? Are you expecting a backing catalog (such as JDBC) to also return either a single snapshot vs all snapshots when the `snapshot-loading-mode` is set?

Reply:
I'm building a Iceberg RESTCatalog server with the ability of CatalogHandlers, and found CatalogHandlers#loadTable doesn't support snapshot queryParams, so I post this PR.

Reply:
@nastra ,  is it necessary to process `snapshot mode` in `CatalogHandlers` ?  if yes, maybe we could add another interface.

Reply:
> I'm building a Iceberg RESTCatalog server with the ability of CatalogHandlers, and found CatalogHandlers#loadTable doesn't support snapshot queryParams, so I post this PR.

In that case wouldn't it be better to implement this functionality on the server rather than in `CatalogHandlers`?

Reply:
ok, I'll implement it on the server,  thanks for your review @nastra